src/main/java/org/sonarlint/intellij/ui/nodes/PrimaryLocationNode.java

```diff
@@ -84,10 +86,12 @@ private String issueCoordinates() {
       return "(-, -) ";
     }
 
-    var doc = rangeMarker.getDocument();
-    var line = doc.getLineNumber(rangeMarker.getStartOffset());
-    var offset = rangeMarker.getStartOffset() - doc.getLineStartOffset(line);
-    return String.format("(%d, %d) ", line + 1, offset);
+    return computeReadActionSafely(() -> {
+      var doc = rangeMarker.getDocument();
+      var line = doc.getLineNumber(rangeMarker.getStartOffset());
+      var offset = rangeMarker.getStartOffset() - doc.getLineStartOffset(line);
+      return String.format("(%d, %d) ", line + 1, offset);
+    });
```

Comment:
Maybe the same for this one, with a `formatRangeMarker` method? We have the same snippet 4 times